src/components/Footer/Footer.js

```diff
@@ -28,8 +39,21 @@ const Footer = ({
   blockStatus,
   loadingVote,
   showScoreButton = false,
-  onPressScore
+  onPressScore,
+  isShowDM = false
 }) => {
+  const {sendMessageDM} = useDMMessage();
+  const [profile] = React.useContext(Context).profile;
+  const [userAllowDm, setUserAllowDm] = React.useState(false);
+
+  const [loading, setLoading] = React.useState({
+    loadingDm: false,
+    loadingDmAnon: false,
+    loadingGetAllowAnonDmStatus: false
+  });
+  const refSheet = React.useRef();
+  const {createSignChat} = useCreateChat();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of multiple state variables and a state object (`loading`) in the same component can lead to confusion. It's better to stick with one approach for consistency. Also, it seems like `loading` is being used to track the loading status of different operations. Instead of using a separate boolean for each operation, consider using an enumeration (string) that represents the current loading operation.

```diff
-  const [loading, setLoading] = React.useState({
-    loadingDm: false,
-    loadingDmAnon: false,
-    loadingGetAllowAnonDmStatus: false
-  });
+  const [loadingOperation, setLoadingOperation] = React.useState(null);
```
Then you can set `setLoadingOperation('dm')`, `setLoadingOperation('dmAnon')`, or `setLoadingOperation('getAllowAnonDmStatus')` as needed. When not loading, set it back to `null`.



<!-- This is an auto-generated comment by OSS CodeRabbit -->